src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1249,233 +1255,1063 @@ subroutine eigencalving(domain, err)
+         call mpas_pool_get_array(geometryPool, 'nstar', nstar)
+         call mpas_pool_get_array(geometryPool, 'damageSource', damageSource)
+         call mpas_pool_get_array(geometryPool, 'damage_nye', damage_nye)
+         call mpas_pool_get_array(geometryPool, 'damageMax', damageMax)
+         call mpas_pool_get_array(geometryPool, 'calvingVelocity', calvingVelocity)
+         call mpas_pool_get_array(geometryPool, 'floatingBasalMassBal', floatingBasalMassBal)
+
+         call mpas_pool_get_field(scratchPool, 'alpha', alphaVar )
+         call mpas_allocate_scratch_field(alphaVar, .true.)
+         alpha => alphaVar % array
+
+         call mpas_pool_get_array(velocityPool, 'eMax', eMax) 
+         call mpas_pool_get_array(velocityPool, 'sigmaxx', sigmaxx)
+         call mpas_pool_get_array(velocityPool, 'sigmayy', sigmayy)
+         call mpas_pool_get_array(velocityPool, 'sigmaxy', sigmaxy)
+         call mpas_pool_get_array(velocityPool, 'sigmayx', sigmayx)
+         call mpas_pool_get_array(velocityPool, 'sigmamax', sigmamax)
+         call mpas_pool_get_array(velocityPool, 'sigmamin', sigmamin)
+
+         where (thickness == 0.0_RKIND)
+             alpha = 0.0_RKIND
+             s0 = 0.0_RKIND
+         elsewhere
+             alpha = sigmamin / sigmamax 
+             s0 = &
+             config_ice_density*(config_ocean_density-config_ice_density)*gravity*thickness/(2.0_RKIND*sigmamax*config_ocean_density)
+         endwhere
+
+         !alpha(:) = sigmamin(:) / (sigmamax(:))
+         !epsmax(:) = sigmamax(:) / (2.0d0*effectiveViscosity(:))
+         !s0(:) =
+         !config_ice_density*(config_ocean_density-config_ice_density)*gravity*thickness(:)/(2.0d0*sigmamax(:)*config_ocean_density)
+          ! Compute the hydrostatic to tensile stress ratio
+          ! These commented lines are doing the same thing as the above where loop. Which is better?
+
+         nstar(:) = &
+         12.0_RKIND*(1.0_RKIND+alpha(:)+alpha(:)**2.0_RKIND)/(4.0_RKIND*(1.0_RKIND+alpha(:)+alpha(:)**2.0_RKIND)+6.0_RKIND*alpha(:)**2.0_RKIND) 
+          ! Compute the effective flow law exponent
+
+         !s0(:) =
+         !config_ice_density*(config_ocean_density-config_ice_density)*gravity*thickness(:)/(2.0d0*sigmamax(:)*config_ocean_density)
+          ! Compute the hydrostatic to tensile stress ratio
+
+         !do iCell = 1, nCells
+         !    if (li_mask_is_floating_ice(cellMask(iCell))) then
+         !        floatingBasalMassBal(iCell) = 0.2/31536000.0*tanh((lowerSurface(iCell)-bedTopography(iCell))/75.0) * max(-100-lowerSurface(iCell),0.0)
+         !    end if
+         !end do
+         ! Temporary change of BMB for the MISMIP+ Ice1r experiment
+
+         do iCell = 1, nCells
+         if (thickness(iCell) == 0.0_RKIND) then
+             damageSource(iCell) = 0.0_RKIND
+         else
+             damageSource(iCell) = nstar(iCell) * (1.0_RKIND-s0(iCell))*eMax(iCell)-floatingBasalMassBal(iCell)/config_ice_density/(thickness(iCell))
```

Comment:
Done.

I added a comment on where this comes from. This is eqn. 26 from Bassis and Ma.

I also added a longer note here about the last part of the expression, which differs from the Bassis and Ma version (because I went on a boondoggle to make sure that gave the correct units -- it does).